source/solvers/heat_transfer_assemblers.cc

```diff
@@ -500,6 +500,63 @@ template class HeatTransferAssemblerRobinBC<2>;
 template class HeatTransferAssemblerRobinBC<3>;
 
 
+template <int dim>
+void
+HeatTransferAssemblerHeatFluxBC<dim>::assemble_matrix(
+  HeatTransferScratchData<dim> &scratch_data,
+  StabilizedMethodsCopyData    &copy_data)
+{
+  (void)scratch_data.is_boundary_cell;
+  (void)copy_data.local_matrix;
+}
+
+template <int dim>
+void
+HeatTransferAssemblerHeatFluxBC<dim>::assemble_rhs(
+  HeatTransferScratchData<dim> &scratch_data,
+  StabilizedMethodsCopyData    &copy_data)
+{
+  if (!scratch_data.is_boundary_cell)
+    return;
+
+  auto &local_rhs = copy_data.local_rhs;
+
+  //  Heat flux Neumann boundary condition, loop on faces
+  for (unsigned int i_bc = 0; i_bc < this->boundary_conditions_ht.size; ++i_bc)
+    {
+      if (this->boundary_conditions_ht.type[i_bc] ==
+          BoundaryConditions::BoundaryType::heat_flux_bc)
+        {
+          const double heat_flux_bc =
+            this->boundary_conditions_ht.heat_flux_bc[i_bc]->value(
+              Point<dim>());
+
+          for (unsigned int f = 0; f < scratch_data.n_faces; ++f)
+            {
+              if (scratch_data.boundary_face_id[f] ==
+                  this->boundary_conditions_ht.id[i_bc])
+                {
+                  for (unsigned int q = 0; q < scratch_data.n_faces_q_points;
+                       ++q)
+                    {
+                      const double JxW = scratch_data.face_JxW[f][q];
+                      for (unsigned int i = 0; i < scratch_data.n_dofs; ++i)
+                        {
+                          const double phi_face_T_i =
+                            scratch_data.phi_face_T[f][q][i];
+                          local_rhs(i) += phi_face_T_i * heat_flux_bc * JxW;
```

Comment:
Shouldn't it be a negative here, and in the parameters you impose the positive value? (application test's value must be changed)
```suggestion
                          local_rhs(i) -= phi_face_T_i * heat_flux_bc * JxW;
```

Reply:
Well it depends right. Because the boundary condition when you have a flux is  -k * grad(T) = q. Consequently, most likley there is (-)*(-) being lumped here. The analytical solution problem will be able to tell you what it is. A positive flux means heat is going out (flux is aligned with the outward normal) and a negative flux means the heat is going in.

Reply:
Using this definition (positive flux means alligned with outward normal and negative means agaisnt it), it means that currently there is indeed a sign mistake. You are specifying a flux of -1000smthing in your prm file. Consequently, we would expect that this means that energy is getting inside of the system (it is not aligned with the outward normal, but on the opposite side). So I think I would clear this out in the documentation (mention that the flux is aligned with the unit normal going out) and that a flux getting in thus has to be negative and a flux going out thus has the be positive. This would make things a lot clearer and would prevent this type of confusion again. It's the issue when you don't have Robin BCs, the convention is not trivial to define.

Reply:
I made it + because I felt it to be more intuitive for the user to do positive if you say heat is going in and negative if heat is going out of the system. Can I just add this explanation to the documentation or you would rather have me changing to take as reference the normal vector (and still add this to the docs)?

Reply:
Well it depends. I guess to me it makes more sense that negative is going in (in a vector sense), but we don't have a lot of things like that in the code so as long as we are clear, it's ok. I would prefer (-) = in and (+) = out, but if other people feel strongly about the other way, I can be flexible. I just prefer the mathematical definition of it lets say

Reply:
I would also prefer (-) = in, but whatever you choose leaving a note that mentions it would be welcome and could save some time for users.

Reply:
(-) it is